SwiftReflector/Inventory/FunctionInventory.cs

```diff
@@ -22,12 +22,14 @@ public override void Add (TLDefinition tld, Stream srcStm)
 			if (tlf == null)
 				throw ErrorHelper.CreateError (ReflectorError.kInventoryBase + 10, $"expected a top-level function but got a {tld.GetType ().Name}");
 
-			OverloadInventory overloads = null;
-			if (!values.TryGetValue (tlf.Name, out overloads)) {
-				overloads = new OverloadInventory (tlf.Name, sizeofMachinePointer);
-				values.Add (tlf.Name, overloads);
+			lock (valuesLock) {
+				OverloadInventory overloads = null;
+				if (!values.TryGetValue (tlf.Name, out overloads)) {
+					overloads = new OverloadInventory (tlf.Name, sizeofMachinePointer);
+					values.Add (tlf.Name, overloads);
+				}
+				overloads.Add (tlf, srcStm);
 			}
```

Comment:
Same, which also means that changing the dictionary in the base type will make all of your subclasses thread-safe.